### PR TITLE
GPG Key handling (#1659)

### PR DESCRIPTION
**Pull request by keichwa**
# Description
* GPG Key handling
* add note about debian based systems
* Update modules/client-configuration/pages/snippets/trust_gpg.adoc
Co-authored-by: Karl Eichwalder <ke@suse.de>
Co-authored-by: Ornela Marić <omaric@suse.com>

https://github.com/uyuni-project/uyuni-docs/pull/1659

# Target branches

Which documentation version does this PR apply to?

- [ ] Master (Default)
- [x] Manager-4.3

# Links

Fixes #<insert issue or PR link, if any>

### CODE REVIEW

File: modules/client-configuration/pages/snippets/trust_gpg.adoc [hunk 1]
```diff
@@ -1,18 +1,61 @@
-By default, operating systems trust only their own GPG keys when they are installed, and do not trust keys provided by third party packages.
+Operating systems either trust their own GPG keys directly or at least ship them installed with the minimal system.
+But third party packages signed by a different GPG key need manual handling.
 The clients can be successfully bootstrapped without the GPG key being trusted.
 However, you cannot install new client tool packages or update them until the keys are trusted.
 
-Salt clients are set to trust {suse} tools channels GPG keys when they are bootstrapped.
-For all other clients and channels, you need to manually trust third party GPG keys.
+Salt clients use now GPG key information entered for a software channel to manage the trusted keys.
```
Comment (jcayouette) [inline modules/client-configuration/pages/snippets/trust_gpg.adoc:6]: use now or now use?

File: modules/client-configuration/pages/snippets/trust_gpg.adoc [hunk 2]
```diff
@@ -1,18 +1,61 @@
...
+When a software channel with GPG key information is assigned to a client, the key gets trusted as soon as the channel is refreshed or the first package gets installed from this channel.
```
Comment (jcayouette) [inline modules/client-configuration/pages/snippets/trust_gpg.adoc:7]: When a software channel with GPG key information is assigned to a client, the key is trusted once the channel is refreshed or the first package is installed from the channel.

File: modules/client-configuration/pages/snippets/trust_gpg.adoc [hunk 3]
```diff
@@ -1,18 +1,61 @@
...
 
+// The meaning of the following sentence is not clear
+The GPG key URL which is set of a software channel must exist.
```
Comment (jcayouette) [inline modules/client-configuration/pages/snippets/trust_gpg.adoc:10]: Sorry confused by this line.

File: modules/client-configuration/pages/snippets/trust_gpg.adoc [hunk 4]
```diff
@@ -1,18 +1,61 @@
...
-If you are bootstrapping Salt clients from the {productname} {webui}, you can use a Salt state to trust the key.
-For more information on custom Salt states, see xref:specialized-guides:salt/salt-custom-states.adoc[].
+Because Debian based systems sign only metadata, there is typically no need to specify extra keys for single channels.
+If a user configures an own GPG key to sign the metadata as described in "Use Your Own GPG Key" in xref:administration:repo-metadata.adoc[] the deployment and trust of that key is executed automatically.
 ====
 
 
+=== User defined GPG keys
+
+Users can define their own GPG keys to be deployed to the client.
```
Comment (jcayouette) [inline modules/client-configuration/pages/snippets/trust_gpg.adoc:26]: Users can define custom GPG keys to be deployed to a client.

File: modules/client-configuration/pages/snippets/trust_gpg.adoc [hunk 5]
```diff
@@ -1,18 +1,61 @@
...
+// I think the ls command is here to show the contents of the directory.
+// In this case, the ls is superfluous.
+
+----
+ls -la /etc/salt/gpg/
+/etc/salt/gpg/my_first_gpg.key
+/etc/salt/gpg/my_second_gpgkey.gpg
+----
+
+The keys are now deployed to the client at [path]``/etc/pki/rpm-gpg/my_first_gpg.key`` and [path]``/etc/pki/rpm-gpg/my_second_gpgkey.gpg``.
```
Comment (jcayouette) [inline modules/client-configuration/pages/snippets/trust_gpg.adoc:52]: The keys are deployed to the client

File: modules/client-configuration/pages/snippets/trust_gpg.adoc [hunk 6]
```diff
@@ -32,5 +75,5 @@ You do not need to delete any previously stored keys.
 ====
 Trusting a GPG key is important for security on clients. 
 It is the task of the admin to decide which keys are needed and can be trusted.
-Trusting the key is done manually,  either by writing a Salt state or adding the keys to the bootstrap script.
+Because a software channel cannot be used when the GPG key is not trusted, the decision of assigning a channel to a client depends on the decision of trusting the key.
```
Comment (jcayouette) [inline modules/client-configuration/pages/snippets/trust_gpg.adoc:78]: A software channel cannot be used when a GPG key is not trusted. Assigning a channel to a client depends on if its GPG key has been trusted.

File: modules/client-configuration/pages/snippets/trust_gpg.adoc (hunk 2 above)
Comment (keichwa) [reply modules/client-configuration/pages/snippets/trust_gpg.adoc:7]: thanks

**Pull request merged**